src/customer/strategies/bolt/bolt-customer-strategy.ts

```diff
@@ -0,0 +1,119 @@
+    }
+
+    deinitialize(): Promise<InternalCheckoutSelectors> {
+        this._boltClient = undefined;
+
+        return Promise.resolve(this._store.getState());
+    }
+
+    continueAsGuest(credentials: GuestCredentials, options?: CustomerRequestOptions): Promise<InternalCheckoutSelectors> {
+        const { methodId } = options || {};
+
+        if (!methodId) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        return this.openCustomBoltCheckoutWidget(credentials.email, methodId)
+            .then(() => this._store.dispatch(this._billingAddressActionCreator.continueAsGuest(credentials, options)))
+            .then(() => this._store.getState());
+    }
+
+    signIn(credentials: CustomerCredentials, options?: CustomerRequestOptions): Promise<InternalCheckoutSelectors> {
+        const { methodId } = options || {};
+
+        if (!methodId) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        return this._store.dispatch(this._customerActionCreator.signInCustomer(credentials, options))
+            .then(() => this.openCustomBoltCheckoutWidget(credentials.email, methodId))
+            .then(() => this._store.getState());
+    }
+
+    signOut(options?: CustomerRequestOptions): Promise<InternalCheckoutSelectors> {
+        return this._store.dispatch(this._customerActionCreator.signOutCustomer(options));
+    }
+
+    signUp(customerAccount: CustomerAccountRequestBody, options?: CustomerRequestOptions): Promise<InternalCheckoutSelectors> {
+        const { methodId } = options || {};
+
+        if (!methodId) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        return this._store.dispatch(this._customerActionCreator.createCustomer(customerAccount, options))
+            .then(() => this.openCustomBoltCheckoutWidget(customerAccount.email, methodId))
+            .then(() => this._store.getState());
+    }
+
+    private openCustomBoltCheckoutWidget(email: string, methodId: string): Promise<InternalCheckoutSelectors> {
+        return new Promise(resolve => {
+            const state = this._store.getState();
+            const paymentMethod = state.paymentMethods.getPaymentMethod(methodId);
+
+            try {
+                if (!this._boltClient) {
+                    throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+                }
+
+                if (!paymentMethod) {
+                    throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+                }
+
+                if (paymentMethod.initializationData?.customContinueFlowEnabled) {
+                    const callbacks = {
+                        close: resolve,
+                    };
+
+                    this._boltClient.openCheckout(email, callbacks)
+                        .then(result => !result && resolve());
```

Comment:
Can `openCheckout` return an error / rejected promise. If so, we need to handle it accordingly,.

Reply:
It is already wrapped in `try...catch`.
I will remove that `then` chaining, because it is unnecessary anymore

Reply:
The `try/catch` block has no effect because you're using the `then` syntax. It only works if you use the `async/await` syntax.